pkg/controller/build/build_monitor.go

```diff
@@ -129,3 +138,28 @@ func monitorRunningBuild(build *v1.Build) {
 func monitorFinishedBuild(build *v1.Build) {
 	runningBuilds.Delete(types.NamespacedName{Namespace: build.Namespace, Name: build.Name}.String())
 }
+
+func scheduledReadyBuildcondition(build *v1.Build) *v1.BuildCondition {
+	return &v1.BuildCondition{
+		Type:   v1.BuildConditionScheduled,
+		Status: corev1.ConditionTrue,
+		Reason: v1.BuildConditionReadyReason,
+		Message: fmt.Sprintf(
+			"the build (%s) is scheduled",
+			build.Name,
+		),
+	}
+}
+
+func scheduledWaitingBuildcondition(build *v1.Build, reason string) *v1.BuildCondition {
```

Comment:
Duplicated code. Either you have a generic func to return a condition, or simply favor the inline creation of the condition from the callers.